src/microcalibrate/calibration.py

```diff
@@ -59,3 +67,62 @@ def calibrate(self) -> None:
 
         self.loss_matrix = self.loss_matrix.loc[subsample]
         self.weights = new_weights
+
+    def _assess_targets(
+        self,
+        loss_matrix: pd.DataFrame,
+        weights: np.ndarray,
+        targets: np.ndarray,
+        target_names: Optional[np.ndarray] = None,
+    ) -> None:
+        """Assess the targets to ensure they do not violate basic requirements like compatibility, correct order of magnitude, etc.
+
+        Args:
+            loss_matrix (pd.DataFrame): DataFrame containing the loss matrix.
+            weights (np.ndarray): Array of original weights.
+            targets (np.ndarray): Array of target values.
+            target_names (np.ndarray): Optional names of the targets for logging. Defaults to None.
+
+        Raises:
+            ValueError: If the targets do not match the expected format or values.
+            ValueError: If the targets are not compatible with each other.
+        """
+        logger.info("Performing basic target assessment...")
+
+        if targets.ndim != 1:
+            raise ValueError("Targets must be a 1D NumPy array.")
+        if len(targets) != loss_matrix.shape[1]:
+            raise ValueError(
+                f"Mismatch: {len(targets)} targets given, but loss_matrix has {loss_matrix.shape[1]} columns."
+            )
+        if np.any(np.isnan(targets)):
+            raise ValueError("Targets contain NaN values.")
+
+        if np.any(targets < 0):
+            logger.warning(
+                "Some targets are negative. This may not make sense for totals."
+            )
+
+        # Estimate order of magnitude from weighted column sums and warn if they are off by an order of magnitude from targets
+        weighted_estimates = weights @ loss_matrix.values
```

Comment:
Let's replace these weights with just 1-weights to make sure we're not flagging this when weights may just start at zero (which is probably common)